gtk-doc: build with meson, etc

worldofpeace opened the pull request:
###### Motivation for this change
Mostly wanted to fix building documentation for `glib` and https://github.com/NixOS/nixpkgs/commit/6f892ad16490156247d9bdadceac8fd5ae874bae fixes what I [noticed](https://github.com/NixOS/nixpkgs/pull/61317#issuecomment-492899556) but it still fails.

See: https://gist.github.com/worldofpeace/7a54fc196ff6ecbe35b11ed2106422ec

Additionally 1.30 added the experimental `gtkdoc-mkhtml2` which needs `anytree` and `lxml`.
I haven't done anything to care for this because adding `anytree` to `gtk-doc` results in one of those
:curly_loop: 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

worldofpeace committed:
gtk-doc: add missing pygments

Also drop six since it was removed in the last release[0]

[0]: https://gitlab.gnome.org/GNOME/gtk-doc/blob/GTK_DOC_1_29/NEWS#L4

worldofpeace committed:
gtk-doc: build with meson

worldofpeace commented on pkgs/development/tools/documentation/gtk-doc/default.nix:
```diff
@@ -1,15 +1,28 @@
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "17h6nwhis66z4dxjrc833wvfl6pqjp81yfx3fq6x7k1qp2749xm4";
+  src = fetchFromGitLab {
```
Some reason the `meson.build` aren't in the `tarball`?

hedning commented:
Looks like we're hitting this: https://gitlab.gnome.org/GNOME/gtk-doc/issues/84
There's a proposed fix: https://gitlab.gnome.org/GNOME/gtk-doc/merge_requests/27

We'll alse need to target staging-next now, since the gtk-doc update have moved there.

worldofpeace committed:
gtk-doc: fix css file permissions

Fixes https://gitlab.gnome.org/GNOME/gtk-doc/issues/84 which
was breaking the documentation build for glib.

The patch was based off an upstream pr [0] but
completed with the requested changes.

[0]: https://gitlab.gnome.org/GNOME/gtk-doc/merge_requests/27

worldofpeace commented:
> Looks like we're hitting this: https://gitlab.gnome.org/GNOME/gtk-doc/issues/84
> There's a proposed fix: https://gitlab.gnome.org/GNOME/gtk-doc/merge_requests/27
> 
> We'll alse need to target staging-next now, since the gtk-doc update have moved there.

Done that, and I've brought a variant of https://gitlab.gnome.org/GNOME/gtk-doc/merge_requests/27 in tree with different permissions.

hedning reviewed (approved):
Lgtm

FRidh commented:
Is this related?
https://hydra.nixos.org/build/93859876

worldofpeace commented:
> Is this related?
> https://hydra.nixos.org/build/93859876

Yes, after reverting all the changes to gtk-doc this builds.
